pkg/resolvers/endpoints.go

```diff
@@ -327,10 +327,14 @@ func (r *defaultEndpointsResolver) getMatchingPodAddresses(ctx context.Context,
 			r.logger.Info("pod IP not assigned yet", "pod", k8s.NamespacedName(&pod))
 			continue
 		}
-		portList := r.getPortList(pod, ports)
-		if len(ports) != len(portList) && len(portList) == 0 {
-			r.logger.Info("Couldn't get matched port list from the pod", "pod", k8s.NamespacedName(&pod), "expectedPorts", ports)
-			continue
+		var portList []policyinfo.Port
+		// no need to populate src pods' ports for ingress rules
+		if rule == networking.PolicyTypeEgress {
+			portList = r.getPortList(pod, ports)
+			if len(ports) != len(portList) && len(portList) == 0 {
```

Comment:
nit: shouldn't this be:
```
if len(portList) == 0 || len(ports) != len(portList) {
   log
   continue
}
```
since we don't want to do anything when port list is empty or port list has any mismatch

Reply:
So, we would want to proceed with the derived port list at the minimum. Say if NP specifies multiple named ports and one of them is not found in the pod spec ..we can setup probes to allow the rest. Now, if nothing is found then we don't want to allow everything for that pod IP

Reply:
Synced offline. Changes were made to the latest update.